### ISSUE TITLE
Build error mkdir: No such file or directory

### ISSUE THREAD

**Issue by jakecraige** [sstephenson/ruby-build#829]
When running `rbenv install 2.2.2` I'm getting a `mkdir` error.

```
~ $ rbenv install 2.2.2 -v
/var/folders/vb/jz3wqx8153vcdhm6dqmd99d80000gn/T/ruby-build.20151114225446.32084 ~
Downloading ruby-2.2.2.tar.gz...
HTTP/1.1 200 OK
Content-Type: binary/octet-stream
Content-Length: 16613636
Connection: keep-alive
Date: Sat, 14 Nov 2015 14:31:44 GMT
Last-Modified: Mon, 13 Apr 2015 20:10:53 GMT
ETag: "326e99ddc75381c7b50c85f7089f3260"
Accept-Ranges: bytes
Server: AmazonS3
Age: 48184
X-Cache: Hit from cloudfront
Via: 1.1 c7d46ec8693394b97aabf8805e5e2f32.cloudfront.net (CloudFront)
X-Amz-Cf-Id: NYPe1mKlI8Mkr7jTnxdDN1AG9ujo7nooCB8BmrMfLfSM9R_c2c-hxg==

-> https://dqw8nmjcqpjn7.cloudfront.net/5ffc0f317e429e6b29d4a98ac521c3ce65481bfd22a8cf845fa02a7b113d9b44
/var/folders/vb/jz3wqx8153vcdhm6dqmd99d80000gn/T/ruby-build.20151114225446.32084/ruby-2.2.2 /var/folders/vb/jz3wqx8153vcdhm6dqmd99d80000gn/T/ruby-build.20151114225446.32084 ~
Installing ruby-2.2.2...
mkdir: : No such file or directory

BUILD FAILED (OS X 10.11.1 using ruby-build 20151028)

Inspect or clean up the working tree at /var/folders/vb/jz3wqx8153vcdhm6dqmd99d80000gn/T/ruby-build.20151114225446.32084
Results logged to /var/folders/vb/jz3wqx8153vcdhm6dqmd99d80000gn/T/ruby-build.20151114225446.32084.log

Last 10 log lines:
Last-Modified: Mon, 13 Apr 2015 20:10:53 GMT
ETag: "326e99ddc75381c7b50c85f7089f3260"
Accept-Ranges: bytes
Server: AmazonS3
Age: 48184
X-Cache: Hit from cloudfront
Via: 1.1 c7d46ec8693394b97aabf8805e5e2f32.cloudfront.net (CloudFront)
X-Amz-Cf-Id: NYPe1mKlI8Mkr7jTnxdDN1AG9ujo7nooCB8BmrMfLfSM9R_c2c-hxg==

/var/folders/vb/jz3wqx8153vcdhm6dqmd99d80000gn/T/ruby-build.20151114225446.32084/ruby-2.2.2 /var/folders/vb/jz3wqx8153vcdhm6dqmd99d80000gn/T/ruby-build.20151114225446.32084 ~
```

The log is pretty unhelpful :/

```
~ $ cat /var/folders/vb/jz3wqx8153vcdhm6dqmd99d80000gn/T/ruby-build.20151114225446.32084.log
/var/folders/vb/jz3wqx8153vcdhm6dqmd99d80000gn/T/ruby-build.20151114225446.32084 ~
HTTP/1.1 200 OK
Content-Type: binary/octet-stream
Content-Length: 16613636
Connection: keep-alive
Date: Sat, 14 Nov 2015 14:31:44 GMT
Last-Modified: Mon, 13 Apr 2015 20:10:53 GMT
ETag: "326e99ddc75381c7b50c85f7089f3260"
Accept-Ranges: bytes
Server: AmazonS3
Age: 48184
X-Cache: Hit from cloudfront
Via: 1.1 c7d46ec8693394b97aabf8805e5e2f32.cloudfront.net (CloudFront)
X-Amz-Cf-Id: NYPe1mKlI8Mkr7jTnxdDN1AG9ujo7nooCB8BmrMfLfSM9R_c2c-hxg==

/var/folders/vb/jz3wqx8153vcdhm6dqmd99d80000gn/T/ruby-build.20151114225446.32084/ruby-2.2.2 /var/folders/vb/jz3wqx8153vcdhm6dqmd99d80000gn/T/ruby-build.20151114225446.32084 ~
```

Any ideas?

**Comment by jakecraige** [comment]
I was able to track this down by going through the script and echoing out stuff. It turns out it was because my `LDFLAGS` were set with spaces between the flag and path. This was in my zshrc:

```
export LDFLAGS="-I /usr/local/include -L /usr/local/lib"
```

Turns out the [substitution here](https://github.com/sstephenson/ruby-build/blob/master/bin/ruby-build#L975) doesn't account for this case so it blows up.

Not sure if I should close or keep this issue open, Feel free to close this issue if it's not something you want to address.

I also ran into other issues caused by other flags being set globally, [this article](http://rubenlaguna.com/blog/2014/04/02/rvm-install-1-dot-9-3-fails-with-configure-error-ansi-c-conforming-const-and-volatile-are-mandatory/) helped.
